### PR TITLE
Shape inference for tile and gather

### PR DESCRIPTION
**Pull request by abhaikollara**

### CODE REVIEW

**Comment by fchollet** [comment]
Please also add corresponding unit tests for it.

**Comment by abhaikollara** [comment]
@fchollet Added test for `gather`. Tests for `tile` were already added.

File: tests/keras/backend/backend_test.py [hunk 1]
```diff
@@ -167,6 +167,26 @@ def test_tile(self):
         if hasattr(th_z, '_keras_shape'):
             assert th_z._keras_shape == th_rep.shape
 
+    def test_gather(self):
+        shape = (10,2,3)
```
Comment (fchollet) [inline tests/keras/backend/backend_test.py]: PEP8 error. Use a PEP8 linter.

File: tests/keras/backend/backend_test.py [hunk 2]
```diff
@@ -167,6 +167,26 @@ def test_tile(self):
...
+        ref = np.arange(np.prod(shape)).reshape(shape)
+        ref_th = KTH.variable(ref)
+        ref_tf = KTF.variable(ref)
+
+        inds = [1,3,7,9]
```
Comment (fchollet) [inline tests/keras/backend/backend_test.py]: PEP8 error. Use a PEP8 linter. It's easy.

Comment (abhaikollara) [inline tests/keras/backend/backend_test.py]: Whoops, keep forgetting about PEP8. Fixing it now :smile:

**Comment by fchollet** [comment]
LGTM, thanks!

**Comment by farizrahman4u** [comment]
Guys, what about bringing this feature back?
#2359

**Pull request merged**